### DISCUSSION TITLE
Clipboard default settings for controlling side - Kind of a security issue

### DISCUSSION THREAD

**Discussion by DomingoMontoya** [rustdesk/rustdesk#4316] [Q&A] [answered]
### Description

Default sharing of clipboard poses a security issue.

As I'm working on a remote computer with RustDesk, everything I copy locally gets sent to the controlled side and vise versa.

### Impact

This can be a big security issue, for everyone, having it that way by default - where sensitive information and passwords end up in the clipboard of a remote client.

If the remote side has a clipboard manager, like CopyQ, then everything the controlling side copies to clipboard is saved on the remote side.

### Additional Context

I would prefer at least the ability to disable clipboard sharing by default. Yes, I can turn it off in each session, but it's too easy to forget.

Preferably, maybe a solution like Mesh Central's where you have either the ability to click a button to send the contents of the clipboard to the remote, or from remote to local, would be great.

MeshCentral also has a 'type' function that I use to avoid the clipboard altogether for passwords.

The type function opens a window, allowing you to input some text, and it's sent as keyboard input to the remote side. That works great for filling in passwords on the remote side, while avoiding having them ending up in the clipboard.

**Comment by rustdesk** [comment, marked as answer]
In nightly build.

![image](https://github.com/rustdesk/rustdesk/assets/71636191/cfa33376-774f-445c-8a12-736565f4cfa6)

**Reply by DomingoMontoya** [reply]
Thanks. I'm using nightly and I tried that but it only disables it on the controlled/remote side, in my testing.

I'm hoping for a way to make sure that, as I connect to and control other remote hosts, my clipboard is never sent to the remote - unless I explicitly enable that feature, or perform some action to send clipboard contents.

**Reply by rustdesk** [reply]
no this setting is for outgoing connections, another one in security page is for incoming connections.

**Reply by DomingoMontoya** [reply]
Doh. I misunderstood. I'm sorry. I didn't notice this wasn't the one under security settings until you said that. Thank you.

**Comment by DomingoMontoya** [comment]
Any more consideration to this? This is a security nightmare. At least on Linux - maybe it works differently on other DE's or OS'es?



# Example:

I'm rustedesked into 4 remote desktops, each belonging to different clients.

I copy a password on my local machine either for my own use, or for client #1.

Every machine I'm connected to with rustdesk ends up with sensitive data in its clipboard that it shouldn't have.

---

I realize that the settings for each client can turn off clipboard for them, but the initiator needs the option to not automatically share clipboard as well, and the security of my local clipboard shouldn't be dictated by the remote's settings.

Currently, if you disable clipboard sharing on rustdesk, it's only for incoming connections. There needs to be an option to disable clipboard for outgoing connections. Even better would be better control as to who the clipboard is shared to when you do share it.

If you look at what MeshCentral has, there are buttons at the bottom of the desktop connection window that allow you to either send or get the remote desktop's current clipboard item. That's the best/safest solution for remote desktop clipboard sharing that I've seen.

**Reply by DomingoMontoya** [reply]
For the record, I just tested this with Anydesk.

Anydesk's default is to share clipboards, but on the initiator (local) side, if I disable clipboard sharing then clipboard sharing is disabled on any outgoing connection I make, which is much safer than the behavior I'm getting currently with rustdesk.

**Reply by DomingoMontoya** [reply]
I also just tested TeamViewer, and it's the same there. If I disable clipboard sharing on my local system, then any remote system I connect to does not get clipboard sync.

**Reply by DomingoMontoya** [reply]
Taking those into consideration, I think this should be moved to bug.
